### PR TITLE
Add Line{Circle|Sky}Region

### PR DESCRIPTION
**Pull request by joleroi**
One step towards fixing https://github.com/astropy/regions/issues/45

### CODE REVIEW

File: regions/shapes/line.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    Parameters
+    ----------
+    start : `~regions.PixCoord`
+        Start position
+    end : `~regions.PixCoord`
+        End position
+    """
+
+    def __init__(self, start, end, meta=None, visual=None):
+        self.start = PixCoord._validate(start, name='start', expected='scalar')
```
Comment (keflavich) [inline regions/shapes/line.py]: Should there be a better way to instantiate `PixCoord`s?  It seems weird to use private methods (`_validate`) here

File: regions/shapes/line.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    Parameters
+    ----------
+    start : `~astropy.coordinates.SkyCoord`
+        Start position
+    end : `~astropy.coordinates.SkyCoord`
+        End position
+    """
+    def __init__(self, start, end, meta=None, visual=None):
+        # TODO: test that start, end is a 0D SkyCoord
+        self.start = start
```
Comment (keflavich) [inline regions/shapes/line.py]: Should we merge without having the `is_coord` check?  Or just add it?

File: regions/shapes/line.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        self.end = end
+        self.meta = meta or {}
+        self.visual = visual or {}
+        self._repr_params = [('start', self.start),
+                             ('end', self.end)]
+
+    @property
+    def area(self):
+        """Region area (`~astropy.units.Quantity`)"""
+        return None
```
Comment (keflavich) [inline regions/shapes/line.py]: `0*u.sr`?

File: regions/shapes/line.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+            start = PixCoord(start_x, start_y)
+            end_x, end_y = skycoord_to_pixel(self.end, wcs=wcs)
+            end = PixCoord(end_x, end_y)
+            return LinePixelRegion(start, end)
+
+        elif mode == 'affine' or mode == 'full':
+            raise NotImplementedError()
+
+        else:
+            raise ValueError('mode should be one of local/affine/full')
```
Comment (keflavich) [inline regions/shapes/line.py]: is mode irrelevant here?

Comment (keflavich) [inline regions/shapes/line.py]: it is not!  All modes do the same.

File: regions/shapes/line.py (hunk 2 above)
Comment (joleroi) [inline regions/shapes/line.py]: This (and also the PixCoord._validate issue) are unrelated to this PR - the same issue is present for all regions at the moment. So I would go for merging this and fixing the input validation separately.

**Comment by joleroi** [comment]
@keflavich , others : Currently the line ``__init__`` takes ``start`` and ``end``. For the parser it would be easier if this was actually a len 2 list of SkyCoords (like a Polygone with 2 vertices). Opinions?

**Pull request merged**